python/GafferUI/WidgetAlgo.py

```diff
@@ -48,3 +54,14 @@ def joinEdges( listContainer ) :
 	for i in range( 0, l ) :
 		visibleWidgets[i]._qtWidget().setProperty( lowProperty, i > 0 )
 		visibleWidgets[i]._qtWidget().setProperty( highProperty, i < l - 1 )
+
+def grab( widget, imagePath ) :
+
+	GafferUI.EventLoop.waitForIdle()
```

Comment:
An optional arg would be even worse to my mind - I'd much rather a hidden hack than an exposed one. I'll buy your "it should just work" argument.